tools/cleanup/cleanup.sh

```diff
@@ -70,22 +76,30 @@ fi
 
 parse_args $@
 
+[[ -z "${PROJECTS}" && -z "${PROJECT_RESOURCE_YAML}" ]] && abort "provide a project or resource file with projects"
+
 (( DRY_RUN )) && echo "-- Running in dry-run mode, no resource deletion --"
-echo "Iterating over projects defined in '${PROJECT_RESOURCE_YAML}', matching '${RE_PROJECT_NAME}"
-target_projects="$(grep -Eio "${RE_PROJECT_NAME}" "${PROJECT_RESOURCE_YAML}")"
-[[ $? -eq 0 ]] || abort "no project found in $PROJECT_RESOURCE_YAML"
+
+if [[ -n "${PROJECT_RESOURCE_YAML}" ]]; then
+  echo "Iterating over projects defined in '${PROJECT_RESOURCE_YAML}', matching '${RE_PROJECT_NAME}"
+  target_projects="$(grep -Eio "${RE_PROJECT_NAME}" "${PROJECT_RESOURCE_YAML}")"
+  [[ $? -eq 0 ]] || abort "no project found in ${PROJECT_RESOURCE_YAML}"
+else
+  target_projects="${PROJECTS}"
+  echo "Iterating over projects [${target_projects} ]"
```

Comment:
what does `[` do?

Reply:
It nicely displays "Iterating over projects [ foo bar ]".